lib/rubygems/executor.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+require 'open-uri'
+require 'json'
+require 'rubygems/platform'
+
+module Gem
+  module Web
+    class Executor
+
+      def supported_options_from_metadata
+        ["source_code_uri", "documentation_uri"]
+      end
+
+      def open_page(gem, options)
+        spec = Gem::Specification.find_by_name(gem)
+
+        if options[:sourcecode]
+          get_info_from_metadata(spec, "source_code_uri")
+        elsif options[:doc]
+          get_info_from_metadata(spec, "documentation_uri")
+        elsif options[:webpage]
+          open_default_browser(spec.homepage)
+        elsif options[:rubygems]
+          open_rubygems(gem)
+        else # The default option is homepage
+          open_default_browser(spec.homepage)
+        end
+      rescue Gem::MissingSpecError => e
+        puts e.message
+      end
+
+      def get_info_from_metadata(spec, info)
+        return unless supported_options_from_metadata.include?(info)
```

Comment:
You need this?

what about:
```ruby
uri = spec.metadata[info]
return unless uri
```

Reply:
I can't return, because this would prevent printing out the error message `Gem '#{spec.name}' does not specify #{info}.`. So I only removed the check